### DISCUSSION TITLE
[Feature] Search by camera (or?) not limited to 250 results

### DISCUSSION THREAD

**Discussion by bcsteeve** [immich-app/immich#3072] [Feature Request]
### The feature

In an attempt to clean up my 70k+ collection, I noticed thousands of photos spread out across the dates that were all taken by a specific camera that I wanted to send to an album.

It was awesome to learn that I could use "m:Cam Name"!  But... in practice, it found exactly 250 results.  I added those to the album and searched again but, of course, it found the same 250 results.  Effectively meaning search is kind of useless for anything with over 250 hits.

Solution 1:  Don't limit the results
Solution 2:  Option to not search within albums

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by bcsteeve** [comment]
Maybe someone could give me a workaround?

I have some 2000+ photos shot on my kid's camera (model is "Kidizoom").  For posterity sake, I want these in my collection, but in an album "kid's camera" or whatever.  

How would I upload these and reasonably add them to an album?  My thought was to upload them, search for m:Kidizoom and that works but is limited to 250 results.  I move those to an album, but searching again gets me those exact same 250 results.  I'm stuck figuring out how to get them into the album other than painstakingly combing through my collection and picking them out.

Ideas?

**Reply by alextran1502** [reply]
Hello, this is the improvement that will be implemented along side with using the same type of view in the main timeline to render virtually any number of assets of the search result. We have plan to implement additional filter among the search result as well

**Reply by bcsteeve** [reply]
Great to hear!

**Reply by jat255** [reply]
@alextran1502 is there an issue/task for this that a user can track to see what the current status of the searching changes are? I'm very interested in these (and would like to help where at all possible) 

P.S. thanks for an amazing piece of software!

**Reply by alextran1502** [reply]
Yes #6968 #6525 and additional work is in the queue. I expect the search enhancement will be in the next release within the next two weeks

**Comment by FlashStopFall** [comment]
Did this feature get canned? 2025, version 1.129.0, and the search is still limited to 250 results. I've uploaded the same data on two accounts and (while it is still not very easy to remove a duplicate image from a second account) the search function is helping me do so. It would go much, much faster though if I could delete more than the 250 selections at a time. (Can this FR be re-opened?)

Loving this software, it is fantastic.

**Reply by bo0tzz** [reply]
The search is not limited, but it is paginated, meaning you'll need to scroll down to load more assets (100 at a time)

**Reply by Jashun44** [reply]
I just started with Immich and came across this issue, and a search lead me here. I am trying to create albums from various search results and realized that the results are paginated. Are there any plans to display all results at once? Or is there an easier way to select all from a search? Scrolling to load more assets is not ideal with thousands of images in a search result. 

Also loving this software!!

**Reply by bo0tzz** [reply]
> Are there any plans to display all results at once? 

This is not feasible. On larger instances it would freeze the browser.

**Reply by Jashun44** [reply]
Got it. Thank you!

**Reply by zsole2** [reply]
Maybe a solution to this could be a field to set the desired number of results in the advanced search (I know, it is already crowded). 
A warning can be added about the drawbacks, so everyone using it would be aware.
The regular search could remain at 250.

Only started using Imminch a few weeks ago, and really-really like it. Good job for all involved !!

**Comment by interunivers** [comment]
I am not sure why but it seems it is broken again a bit. I have a few thousands photos taken by my phone I want to search for but it only reaches back when I search for the camera model about 2 years. I checked again if the model tag changed on older photos but it doesn’t the search results just stop. And also waiting a good amount of time doesn’t load more.
